cfme/base/login.py

```diff
@@ -51,35 +73,29 @@ def current_fullname(self):
 
     @property
     def current_groupname(self):
-        # TODO: Move these locators and accessors to a widget "Group Dropdown" widget
         # 5.9 Locators for finding current group when user has multiple groups
-        current_group_locator = '{}/ul/li/a[@title="Currently Selected Group"]'.format(
-            self.group_list_locator)
         current_group_marker = ' (Current Group)'
-        try:
-            current_group = self.browser.element(current_group_locator)
-            return self.browser.text(current_group).replace(current_group_marker, '')
-        except NoSuchElementException:
-            return self.settings.items[1].strip()
+        current_groups = self.settings.groups.items
+
+        for group in current_groups:
+            if current_group_marker in group:
+                return group.replace(current_group_marker, '')
+        else:
+            # User is only assigned to one group
+            return group
 
     @property
     def group_names(self):
         """ Return a list of the logged in user's assigned groups.
 
         Returns:
-            Version >= 5.9 - list of all groups the logged in user is assigned to
+            Version >= 5.9 - list containing all groups the logged in user is assigned to
             Version < 5.9 - single item list containing the user's current group
         """
-        # TODO: Move these locators and accessors to a widget "Group Dropdown" widget
-        group_list_locator = '{}/ul/li'.format(self.group_list_locator)
         current_group_marker = ' (Current Group)'
```

Comment:
**REQUIRED:** This should be a constant at the top of the module.   It's used at least twice in different functions.

Reply:
I moved it to the SettingsGroupSubMenu class